src/ar_serial_client.c

```diff
@@ -0,0 +1,277 @@
+	// something to revisit
+
+	// canonical mode
+	t->c_cflag &= ~ICANON; // disable
+//	t->c_cflag |= ICANON; // enable
+
+	// echo
+	t->c_cflag &= ~ECHO; // disable
+//	t->c_cflag |= ECHO; // echo
+	
+	// erasure
+	t->c_cflag &= ~ECHOE; // disable
+//	t->c_cflag |= ECHOE; // enable
+
+	// new line echo
+	t->c_cflag &= ~ECHONL; // disable
+//	t->c_cflag |= ECHONL; // enable
+
+	// disable signal chars
+	// this should also probably not be modified
+	t->c_cflag &= ~ISIG;
+
+	// software flow control
+	// I am not totally sure what this does but it seems like we
+	// want it for serial communication according to this doc:
+	// http://uw714doc.sco.com/en/SDK_sysprog/_TTY_Flow_Control.html
+	t->c_iflag &= ~(IXON | IXOFF | IXANY);
+
+	// special byte handling
+	// this should probably not be modified for serial
+	t->c_iflag &= ~(IGNBRK | BRKINT | PARMRK | ISTRIP | INLCR | IGNCR | ICRNL);
+
+	// output modes
+	// these handle character conversion, so just disable them
+	t->c_oflag &= ~OPOST;
+	t->c_oflag &= ~ONLCR;
+
+	// TO DO
+	// VMIN and VTIME are going to be pretty depended on the speed of the DAQ
+	// as far as I know so this needs to be revisited
+	
+	// timing
+	t->c_cc[VTIME] = 5; // TEMP VALUE
+	t->c_cc[VMIN] = 0; // TEMP VALUE
+}
+
+//-------------------------------------------------------------- 
+// Function: read_serial_data
+// Purpose: currently read and display data, this may be changed
+// in the future though
+//--------------------------------------------------------------
+void read_serial_data(int s_port, char dest[BUFFER]){
+	int bytes_received = 0;
+	char read_buffer [256];
+
+	// clear buffer
+	memset(&read_buffer, '\0', sizeof(read_buffer));
+	// read data in from serial port
+	bytes_received = read(s_port, &read_buffer, sizeof(read_buffer));
+	// check bytes recieved
+	if(bytes_received < 0){
+		display_errno_error(errno, "Error reading in read_serial_data");
+	}
+
+	printf("Read %i bytes from serial port. ", s_port);
+	printf("Message recieved: %s\n", read_buffer);
+	memcpy(dest, read_buffer, BUFFER); 
+}
+
+
+//-------------------------------------------------------------- 
+// MAIN
+//--------------------------------------------------------------
+
+int main(int argc, char* argv[]){
+	// serial variables
+	int serial_read_port = 0;
+	int count = 0;
+
+	// socket variables
+	int sock_fd = 0;
+	int tcp_write_port = 0;
+	int message;
+	struct sockaddr_in server_address;
+	struct hostent *server;
+	char* host;
+	char buff[BUFFER];	
+
+	// command line check
+	if(argc != 3){
+		fprintf(stderr, "usage: %s <hostname> <port>\n", argv[0]);
+		exit(0);
+	}
+
+//-------------------------------------------------------------- 
+// SERIAL SETUP BEGIN
+//--------------------------------------------------------------
+	// try to open the port and check for errors	
+	serial_read_port = open(OPEN_PORT, O_RDWR);
+	// attempt to open port 20 times before quiting
+	while(serial_read_port <= 0 && count < 20){
```

Comment:
in practice i would not be surprised if people open the app on the hololens then turn on this program to start sending data. in that case, retrying for a few minutes might be best